engine/access/rest/websockets/data_providers/factory.go

```diff
@@ -0,0 +1,97 @@
+package data_providers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/engine/access/state_stream"
+)
+
+// Constants defining various topic names used to specify different types of
+// data providers.
+const (
+	EventsTopic              = "events"
+	AccountStatusesTopic     = "account_statuses"
+	BlocksTopic              = "blocks"
+	BlockHeadersTopic        = "block_headers"
+	BlockDigestsTopic        = "block_digests"
+	TransactionStatusesTopic = "transaction_statuses"
+)
+
+// DataProviderFactory defines an interface for creating data providers
+// based on specified topics. The factory abstracts the creation process
+// and ensures consistent access to required APIs.
+type DataProviderFactory interface {
+	// NewDataProvider creates a new data provider based on the specified topic
+	// and configuration parameters.
+	//
+	// No errors are expected during normal operations.
+	NewDataProvider(context.Context, string, map[string]string, chan<- interface{}) (DataProvider, error)
```

Comment:
Let's add some names to arguments so that the code more readable and understandable. 
e.g. What is `map[string]string` let's name this arg, or even introduce a type for this?

Reply:
[c5a9f17](https://github.com/onflow/flow-go/pull/6636/commits/c5a9f17c216247068050820f927f38c479ee4d69)